### ISSUE TITLE
Pod Annotations does not be injected in workers when using KubernetesExecutor

### ISSUE THREAD

**Issue by brcps12** [apache/airflow#21386]
### Official Helm Chart version

1.4.0 (latest released)

### Apache Airflow version

2.2.3 (latest released)

### Kubernetes Version

1.21.6

### Helm Chart configuration

```yaml
executor: KubernetesExecutor
workers:
  podAnnotations:
    annotation-name: some value
```

### What happened

Annotations in `workers.podAnnotations` are not injected in workers when executor is `KuberentesExecutor`

### Anything else

In `chart/files/pod-template-file.kubernetes-helm-yaml`, only `.Values.airflowPodAnnotations` is injected.

```yaml
# chart/files/pod-template-file.kubernetes-helm-yaml
apiVersion: v1
kind: Pod
metadata:
  {{- if .Values.airflowPodAnnotations }}
  annotations:
  {{- toYaml .Values.airflowPodAnnotations | nindent 4 }}
  {{- end }}
```

Is this for intended? I think it also needs to insert `podAnnotations` like this:

```yaml
# chart/files/pod-template-file.kubernetes-helm-yaml
apiVersion: v1
kind: Pod
metadata:
  {{- if or (.Values.airflowPodAnnotations) (.Values.workers.podAnnotations) }}
  annotations:
    {{- if .Values.airflowPodAnnotations }}
    {{- toYaml .Values.airflowPodAnnotations | nindent 4 }}
    {{- end }}
    {{- if .Values.workers.podAnnotations}}
    {{- toYaml .Values.workers.podAnnotations | nindent 4 }}
    {{- end }}
  {{- end }}
```

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/apache/airflow/blob/main/CODE_OF_CONDUCT.md)

**Comment by sawaca96** [comment]
kubernetesExecutor can't use worker. isn't it?
```
Error: UPGRADE FAILED: execution error at (airflow/templates/_helpers/validate-values.tpl:43:5): 
If `airflow.executor=KubernetesExecutor`, then all of [`workers.enabled`, `flower.enabled`, `redis.enabled`] should be `false`!
```

**Comment by brcps12** [comment]
Yes, if the `KubernetesExecutor` is used, it seems true that I can’t use the Airflow Worker Pod. However, if set the `executor` to `KubernetesExecutor`, I can specify the template of the pod in which each task will be executed, using `.Values.podTemplate`, and if not specified, `chart/files/pod-template-file.kubernetes-helm-yaml` is used as the default value.

The document describes the `.Values.podTemplate` as follows.

> The contents of `pod_template_file.yaml` used for `KubernetesExecutor` workers (templated). The default (see `files/pod-template-file.kubernetes-helm-yaml`) already takes into account normal `workers` configuration parameters (e.g. `workers.resources`), so you normally won’t need to override this directly.

It says that `workers` configurations are already taken into account, so I think not only `resources` but also `podAnnotations` should be considered.

**Comment by aru-trackunit** [comment]
Still happening on 1.6.0 :(

**Comment by aru-trackunit** [comment]
@potiuk Can you please confirm that this is a valid issue? I have also faced this error and after doing some workarounds I feel like is worth contributing back to the community.

**Comment by potiuk** [comment]
I have no idea why you are calling me - rather avoid doing it if you have no good reason to believe somene has not been involved. 

Simply check it. If it is still there is not, but if you want to fix it you will have to anyhow reproiduce it, so maybe start with it.

**Comment by amoghrajesh** [comment]
@brcps12 @potiuk this was fixed by the PR: #24647
We should close this issue as fixed.
